File: acs-web/AcsStatsWeb/Models/MatchDetails.cs
namespace AcsStatsWeb.Models;

public class MatchDetails
{
    public required string HomeTeam { get; set; }
    public required string AwayTeam { get; set; }
    public required string ResultString { get; set; }
    public required string DurationType { get; set; }
    public required string Tournament { get; set; }
    public required string MatchType { get; set; }
    public required int? HomeTotal1 { get; set; }
    public required int? HomeWickets1 { get; set; }
    public required string? HomeOvers1 { get; set; }
    public required bool? HomeDeclared1 { get; set; }
    public required int? AwayTotal1 { get; set; }
    public required int? AwayWickets1 { get; set; }
    public required string? AwayOvers1 { get; set; }
    public required bool? AwayDeclared1 { get; set; }
    public required int? HomeTotal2 { get; set; }
    public required int? HomeWickets2 { get; set; }
    public required string? HomeOvers2 { get; set; }
    public required bool? HomeDeclared2 { get; set; }
    public required int? AwayTotal2 { get; set; }
    public required int? AwayWickets2 { get; set; }
    public required string? AwayOvers2 { get; set; }
    public required bool? AwayDeclared2 { get; set; }
}

File: acs-api/src/main/kotlin/com/knowledgespike/feature/frontpage/domain/model/MatchDetails.kt
package com.knowledgespike.feature.frontpage.domain.model

import kotlinx.serialization.Serializable

@Serializable
data class MatchDetails(
    val homeTeam: String,
    val awayTeam: String,
    val resultString: String,
    val durationType: String,
    val tournament: String,
    val matchType: String,
    val homeTotal1: Int?,
    val homeWickets1: Int?,
    val homeOvers1: String?,
    val homeDeclared1: Boolean?,
    val awayTotal1: Int?,
    val awayWickets1: Int?,
    val awayOvers1: String?,
    val awayDeclared1: Boolean?,
    val homeTotal2: Int?,
    val homeWickets2: Int?,
    val homeOvers2: String?,
    val homeDeclared2: Boolean?,
    val awayTotal2: Int?,
    val awayWickets2: Int?,
    val awayOvers2: String?,
    val awayDeclared2: Boolean?,
)
